cloudflare/data_source_load_balancer_pools.go

```diff
@@ -0,0 +1,141 @@
+package cloudflare
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceCloudflareLoadBalancerPools() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceCloudflareLoadBalancerPoolsRead,
+
+		Schema: map[string]*schema.Schema{
+			"pools": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
```

Comment:
I hesitated to directly use `resourceCloudflareLoadBalancerPool().Schema` here, although it doesn't offer the `id` field.

Let me know if you think otherwise.

Reply:
why not append the `id` if you need it?

```go
func dataSourceLoadBalancerPoolSchema() map[string]*schema.Schema {
	schema := resourceCloudflareLoadBalancerPool().Schema

	schema["id"] = *schema.Schema{
		Type:     schema.TypeString,
                Computed: true,
 
	}

	return schema
}

func dataSourceCloudflareLoadBalancerPools() *schema.Resource {
	return &schema.Resource{
		Read: dataSourceCloudflareLoadBalancerPoolsRead,
		Schema: dataSourceLoadBalancerPoolSchema(),
        }
}
```

to be honest though, i'm not sure why you'd need anything other than the `id` on it's own. what is your use case for having everything you've listed?

Reply:
actually, no, you don't want the `resourceCloudflareLoadBalancerPool` schema as the fields won't line up correctly as a data source 🤦‍♂️ it's geared towards being a managed resource as opposed to a data source.

Reply:
> what is your use case for having everything you've listed?

We manage pools outside of Terraform (because of automation). But load balancers are in Terraform, so to avoid Terraform wiping them away we need that kind of data source.

We do some other filtering later within Terraform, unfortunately, that's not possible with the API.

Reply:
tbh that would be weird having a data source that does not provide all the information about a pool; different usecases miight need different information from the pools, so might as well have the complete data source right away! (even if it's bound to evolve with lb pools evolutions)